tlsproxy.go

```diff
@@ -27,20 +29,7 @@ func RunClient(l net.Listener, forwardAddr string, keepAlivePeriod time.Duration
 	tlsConfig.ClientSessionCache = tls.NewLRUClientSessionCache(5000)
 
 	doRun(wrapKeepAliveListener(keepAlivePeriod, l), func() (net.Conn, error) {
-		_conn, err := dial(keepAlivePeriod, forwardAddr)
-		if err != nil {
-			return _conn, err
-		}
-		conn := tls.Client(_conn, tlsConfig)
-		err = conn.Handshake()
-		if err != nil {
-			_conn.Close()
-			return nil, err
-		}
-		if !conn.ConnectionState().DidResume {
-			log.Debug("Connection did not resume")
-		}
-		return conn, nil
+		return tls.DialWithDialer(dialer(keepAlivePeriod), "tcp", forwardAddr, tlsConfig)
```

Comment:
It looks like the above was for debugging purposes, but I'm assuming we're good on that now. This just simplifies the code a bit.